src/docfx/config/Config.cs

```diff
@@ -146,97 +146,111 @@ public static Config Load(string docsetPath, CommandLineOptions options, bool ex
         {
             var configPath = Path.Combine(docsetPath, "docfx.yml");
             var exists = File.Exists(configPath);
-            config = exists ? LoadCore(docsetPath, configPath, options, extend, restoreMap) : new Config();
+            if (exists)
+            {
+                var (loadErrors, value) = LoadCore(docsetPath, configPath, options, extend, restoreMap);
+                errors.AddRange(loadErrors);
+                config = value;
+            }
+            else
+            {
+                config = new Config();
+            }
             return exists;
         }
 
-        private static Config LoadCore(string docsetPath, string configPath, CommandLineOptions options, bool extend, RestoreMap restoreMap)
+        private static (List<Error>, Config) LoadCore(string docsetPath, string configPath, CommandLineOptions options, bool extend, RestoreMap restoreMap)
         {
             // Options should be converted to config and overwrite the config parsed from docfx.yml
+            var errors = new List<Error>();
             Config config = null;
-            try
-            {
-                var optionConfigObject = options?.ToJObject();
-                var configObject = ExpandAndNormalize(JsonUtility.Merge(LoadConfigObject(docsetPath, configPath, extend, restoreMap), optionConfigObject));
-                config = configObject.ToObject<Config>(JsonUtility.DefaultDeserializer);
-            }
-            catch (Exception e)
-            {
-                throw Errors.InvalidConfig(configPath, e.Message).ToException(e);
-            }
+            var optionConfigObject = options?.ToJObject();
 
-            Validate(config, docsetPath);
+            var (loadErrors, value) = LoadConfigObject(docsetPath, configPath, extend, restoreMap);
+            errors.AddRange(loadErrors);
 
-            return config;
+            var configObject = ExpandAndNormalize(JsonUtility.Merge(value, optionConfigObject));
+            var deserializeErrors = new List<Error>();
+
+            (deserializeErrors, config) = JsonUtility.ToObject<Config>(configObject);
+            errors.AddRange(deserializeErrors);
+
+            errors.AddRange(Validate(config, docsetPath));
+            return (errors, config);
         }
 
-        private static void Validate(Config config, string docsetPath)
+        private static List<Error> Validate(Config config, string docsetPath)
         {
-            ValidateLocale(config);
-            ValidateContributorConfig(config.Contribution, docsetPath);
+            var errors = new List<Error>();
+            errors.AddRange(ValidateLocale(config));
+            errors.AddRange(ValidateContributorConfig(config.Contribution, docsetPath));
+            return errors;
         }
 
-        private static void ValidateLocale(Config config)
+        private static List<Error> ValidateLocale(Config config)
```

Comment:
can this be implemented as an DataAnnotation extension, like `LocaleAttribute` ?

Reply:
Will create a separate PR for this

---
In reply to: [211462516](https://github.com/dotnet/docfx/pull/3243#discussion_r211462516) [](ancestors = 211462516)